### DISCUSSION TITLE
[It is a bug of my prototype. Please delete this thread]

### DISCUSSION THREAD

**Discussion by lanyusan** [konvajs/konva#1228] [Q&A]
Found out it is a bug of my prototype. Please delete it. Thanks

===

I am continuing building my prototype with Konva. I found a problem with group and clipping. Not sure if i was doing it right or I missed a trick.

My drawing UI needs to support drawing  rect on top of container rect, which is a visual container of shape groups. 

A Konva group has been created with the container rect's area as clipping area.

After a new rect is drawn, which is partly inside the containing rect, and the new rect is then added to the group, the clipping doesn't take effect. 

Am I missing anything? I tried to call `layer.draw / layer.batchDraw / group.draw` etc., the clipping still doesn't work for the new added rect.

**Comment by lavrton** [comment]
Make a small demo, please.
